libgadget/exchange.c

```diff
@@ -45,10 +47,10 @@ typedef struct {
  * exchange particles according to layoutfunc.
  * layoutfunc gives the target task of particle p.
 */
-static int domain_exchange_once(ExchangePlan * plan, int do_gc);
+static int domain_exchange_once(ExchangePlan * plan, int do_gc, MPI_Comm ExchangeComm);
```

Comment:
Probably just call it comm, since the routine takes exactly one comm and it is for exchange. `MPI_Bcast` doesn't take a BcastComm, for example. It can reduce effort rewriting future code following this style.

Reply:
Fixed!